src/dataframerow/utils.jl

```diff
@@ -94,6 +94,59 @@ isequal_row(cols1::Tuple{Vararg{AbstractVector}}, r1::Int,
     isequal(cols1[1][r1], cols2[1][r2]) &&
         isequal_row(Base.tail(cols1), r1, Base.tail(cols2), r2)
 
+# Simple vector type for internal use which represents a virtual DataAPI.refpool
+# so that a vector of reals can be its own DataAPI.refarray:
+# this just allows telling the generic row_group_slots method for generic refpools
+# what is the minimum index and the number of (potential) groups
+# missing values are supported and represented used index max+1
+struct IntegerRefpool{T<:Union{Real, Missing}} <: AbstractVector{T}
```

Comment:
we should disallow bare `Missing` I think

Reply:
or not? (I just started reading this PR 😄 )

Reply:
Also `Union{}` meets the restriction you specified.

Reply:
I don't think that's a problem. I'll add tests.

Reply:
OK - it is just my second reflex that in such `Union` cases `Missing` and `Union{}` are corner case that should be considered. If it is not a problem here then it is great.